src/Manipulator/ServicesManipulator.php

```diff
@@ -48,7 +45,7 @@ public function __construct(string $file)
      *
      * @throws \RuntimeException
      */
-    public function addResource(string $serviceId, string $modelClass, string $adminClass, string $controllerName, string $managerType): void
+    public function addResource(string $modelClass, string $adminClass, string $controllerName, string $managerType): void
```

Comment:
The idea is good, but changing the signature is a BC-break.
If someone is doing 
```
$servicesManipulator->addResource(
                $id,
                $modelClass,
                $adminClassFullName,
                $controllerName,
                $managerType
            );
```
it won't work anymore.

So I would say you have to keep the `id` param and add a `NEXT_MAJOR: Remove the id param`

Reply:
I guess it's used just to genera admin configuration in service.yaml

And it was used just in Make folder...


Do you think someone is using this method directly in the code? Specially to sonata 4?

Reply:
I would say that I dont think it's used by someone. But the BC policy doesnt allow us to take this "risk".

But maybe should be start to declare some code internal.

Because for exemple, changing the template of the service generation is technically a Bc break, but it's understandable to change it sometimes.

Wdyt @jordisala1991 ?

Reply:
Agree, if the class is not internal we should respect the bc promise.

Reply:
Ok, I'll made this change then , it'll be like this

```diff
admin.category:
        class: App\Admin\CategoryAdmin
        tags:
            - { name: sonata.admin, model_class: App\Entity\Category, controller: ~, manager_type: orm, label: Category }
-        public: true
```

and add it as `@internal`

Question: do you know someway to deprecate this service_id parameter?

or should I create a new method and deprecate the old one? any idea?

Reply:
We can do it this way.
- Changing the signature of the method from 5 param to 4 param.
- Using a func_get_args or func_num_args check at the beginning of the method
- If 5 param are passed, we trigger a deprecation with the old behavior
- If 4 param are passed, we use the new behavior.

But in this situation, I think it could be still useful to have the id param.

People could prefer
```
admin.category: 
    class: App\Admin\CategoryAdmin
    tags: - { name: sonata.admin, model_class: App\Entity\Category, controller: ~, manager_type: orm, label: Category }
```
over
```
App\Admin\CategoryAdmin
     tags: - { name: sonata.admin, model_class: App\Entity\Category, controller: ~, manager_type: orm, label: Category }
```

So I would provide a way to have both depending on the `$input->getOption('id')` value provided in the AdminMaker

Reply:
> So I would provide a way to have both depending on the $input->getOption('id') value provided in the AdminMaker

do you like: `do you want to provider service_id? (yes)`

Reply:
But I was thinking isn't it possible create this configuration using automatic?

like this: https://symfony.com/doc/current/service_container/tags.html#autoconfiguring-tags

and for dynamically parameters we could get from that admin like
- controller
- modelClass
- label
- group

Reply:
Do we need to add an option ?

To me, there is already
```
->addOption('id', 'i', InputOption::VALUE_OPTIONAL, 'The admin service ID');
```

We could say that
- If an id is provided, it's used
- If not, we use the class directly

Reply:
> We could say that
> 
> If an id is provided, it's used
> If not, we use the class directly

nice, it's good for me :)

Reply:
> But I was thinking, isn't it possible create this configuration automatic?
> using this: https://symfony.com/doc/current/service_container/tags.html#autoconfiguring-tags
> and for dynamically parameters we could get from the admin like

Sure ! 
There is an issue about it https://github.com/sonata-project/SonataAdminBundle/issues/6551
And some tries https://github.com/sonata-project/SonataAdminBundle/pull/7660
It could be configured with static methods.

Reply:
> > But I was thinking, isn't it possible create this configuration automatic?
> > using this: https://symfony.com/doc/current/service_container/tags.html#autoconfiguring-tags
> > and for dynamically parameters we could get from the admin like
> 
> Sure ! There is an issue about it #6551 And some tries #7660 It could be configured with static methods.

I will check that ;)